app/src/main/java/com/arctouch/codechallenge/home/HomeActivity.kt

```diff
@@ -1,30 +1,115 @@
 package com.arctouch.codechallenge.home
 
+import android.annotation.SuppressLint
+import android.content.Intent
 import android.os.Bundle
+import android.os.Handler
+import android.support.v7.app.AppCompatActivity
+import android.support.v7.widget.LinearLayoutManager
+import android.support.v7.widget.RecyclerView
 import android.view.View
+import android.widget.TextView
 import com.arctouch.codechallenge.R
+import com.arctouch.codechallenge.api.RetrofitConfig.Companion.api
 import com.arctouch.codechallenge.api.TmdbApi
-import com.arctouch.codechallenge.base.BaseActivity
 import com.arctouch.codechallenge.data.Cache
+import com.arctouch.codechallenge.detail.DetailMovieActivity
+import com.arctouch.codechallenge.listener.InfiniteScrollListener
+import com.arctouch.codechallenge.model.Movie
+import com.arctouch.codechallenge.util.verifyAvailableNetwork
 import io.reactivex.android.schedulers.AndroidSchedulers
 import io.reactivex.schedulers.Schedulers
 import kotlinx.android.synthetic.main.home_activity.*
+import java.io.Serializable
 
-class HomeActivity : BaseActivity() {
+class HomeActivity : AppCompatActivity(), HomeAdapter.OnItemListener, InfiniteScrollListener.OnLoadMoreListener {
+
+    lateinit var moviesWithGenres: MutableList<Movie>
+    lateinit var connectionText: TextView
+    lateinit var recyclerView: RecyclerView
+    lateinit var manager: LinearLayoutManager
+    lateinit var infiniteScrollListener: InfiniteScrollListener
+    lateinit var homeAdapter: HomeAdapter
+    var pageCounter: Long = 0
+    var position: Int = 0
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.home_activity)
 
-        api.upcomingMovies(TmdbApi.API_KEY, TmdbApi.DEFAULT_LANGUAGE, 1, TmdbApi.DEFAULT_REGION)
-            .subscribeOn(Schedulers.io())
-            .observeOn(AndroidSchedulers.mainThread())
-            .subscribe {
-                val moviesWithGenres = it.results.map { movie ->
-                    movie.copy(genres = Cache.genres.filter { movie.genreIds?.contains(it.id) == true })
+        initViews()
+
+        if (verifyAvailableNetwork(this)) {
```

Comment:
The Android documentation says: "As a user navigates through, out of, and back to your app, the Activity instances in your app transition through different states in their lifecycle. The Activity class provides a number of callbacks that allow the activity to know that a state has changed: that the system is creating, stopping, or resuming an activity, or destroying the process in which the activity resides." And continues: "To navigate transitions between stages of the activity lifecycle, the Activity class provides a core set of six callbacks: onCreate(), onStart(), onResume(), onPause(), onStop(), and onDestroy()." 

This validation isn't a good practice because sometimes the network connection can be turned off or even can ocorrs a temporary loss of connection and this validation may not detect it. The documentation recommends implements the use of BroadcastReceiver to detect any connection changes in order to avoid any crashing.